firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3469,12 +3483,14 @@ extension BrowserViewController: TabManagerDelegate {
     func updateTabCountUsingTabManager(_ tabManager: TabManager, animated: Bool = true) {
         if let selectedTab = tabManager.selectedTab {
             let count = selectedTab.isPrivate ? tabManager.privateTabs.count : tabManager.normalTabs.count
-            if isToolbarRefactorEnabled {
-                let action = ToolbarAction(numberOfTabs: count,
-                                           windowUUID: windowUUID,
-                                           actionType: ToolbarActionType.numberOfTabsChanged)
+            if isToolbarRefactorEnabled,
+               let toolbarState = store.state.screenState(ToolbarState.self, for: .toolbar, window: windowUUID),
+               toolbarState.numberOfTabs != count {
```

Comment:
nit: maybe add private func that has a name that clearly describes more why we are sending this call when `numberOfTabs != count`

Reply:
same here regarding tests